benchexec/tools/divine.py

```diff
@@ -87,15 +97,15 @@ def determine_result(self, returncode, returnsignal, output, isTimeout):
         (e.g., "CRASH", "OUT_OF_MEMORY", etc.).
         """
 
-        if not output:
+        if not run.output:
             return "ERROR - no output"
 
-        last = output[-1]
+        last = run.output[-1]
 
-        if isTimeout:
+        if run.was_timeout:
             return "TIMEOUT"
 
-        if returncode != 0:
+        if run.exit_code.value != 0:
```

Comment:
Note that this changes behavior compared to the old code. Previously , `returncode` was 0 if the tool was killed by a signal, e.g., due to a timeout, now it is `None`. Is this change intended?

Reply:
`None != 0` also evaluates to true

Reply:
Exactly, this is why there is a difference to the previous behavior. The branch will now be entered if `value` is `None`, previously it was not entered if `value` was 0.

Reply:
made it similar to others in f2d7b96e04126b075a70bf66ecc07b64b0be151e

Reply:
I don't think the current code is the intended way for this tool-info module (and behaves differently than before). For OOMs, it would return `ERROR - Pre-run` instead of `UNKNOWN` as before.